### ISSUE TITLE
ML Vision dependencies

### ISSUE THREAD

**Issue by shliama** [brianmtully/flutter_google_ml_vision#9]
Hey, thanks for the plugin 🙌

I have a question regarding the **Android** & **iOS** dependencies, in the `firebase_ml_vision` plugin we had to manually specify whatever we want to use in either `build.gradle` or `Podfile`. 

Here, I see these dependencies specified inside `android/build.gradle`:
```
implementation 'com.google.mlkit:face-detection:16.0.6'
implementation 'com.google.mlkit:barcode-scanning:16.1.1'
implementation 'com.google.mlkit:image-labeling:17.0.3'
implementation 'com.google.mlkit:object-detection:16.2.3'
implementation 'com.google.android.gms:play-services-mlkit-text-recognition:16.1.3'
implementation 'com.google.mlkit:language-id:16.1.1'
```

And these inside `ios/google_ml_vision.podspec`:
```
s.dependency 'GoogleMLKit/BarcodeScanning'
s.dependency 'GoogleMLKit/FaceDetection'
s.dependency 'GoogleMLKit/ImageLabeling'
s.dependency 'GoogleMLKit/TextRecognition'
```

Does it mean all these binaries are being packaged inside iOS/Android apps, even if I just want to use only Barcode scanning?

**Comment by brianmtully** [comment]
Yes that is correct. I can remove those dependencies to allow you to only use the features you need.

**Comment by shliama** [comment]
I think it's a way to go, let me know if you need any help with the changes.

**Comment by festapp86** [comment]
Yes, we also would only need TextDetection. The final binary apk file is quite big if all dependencies are being packaged.
Do you have a rough timeline, when this is available? Thanks

**Comment by festapp86** [comment]
@shliama did you find a way to only compile the libs you actually need?

**Comment by shliama** [comment]
Not really. I haven't had time to research this yet, but most probably we should look at how it was done with the [firebase_ml_vision](https://pub.dev/packages/firebase_ml_vision) plugin.

**Comment by festapp86** [comment]
@brianmtully we would highly appreciate a timeline. Thanks

**Comment by swissonid** [comment]
Hi, @brianmtully is spent yesterday sketching and implement a [possible solution](https://github.com/swissonid/flutter_google_ml_vision/tree/bugfix/%239-ML-Vision-dependencies) for this problem. The changes are going to be big and I do not know if it allies with your vision of this plugin. 

The idea behind it is to split the plugin into multiple packages like the [Firebase packages](https://github.com/FirebaseExtended/flutterfire).

What do you think @brianmtully ?

**Comment by AhmedMahhmoud** [comment]
I am only using face detection splitting it would be good for app size since it is now around 70m with only the detection being used !
